src/compiler/checker.ts

```diff
@@ -2452,6 +2452,69 @@ module ts {
             return result;
         }
 
+        function getBaseTypes(type: InterfaceType): ObjectType[] {
+            if (!(<InterfaceTypeWithBaseTypes>type).baseTypes) {

```

Comment:
I can use a local. The problem with ResolvedInterfaceType is that it implies that the type elements (members, call signatures, etc) are resolved, and that is not in fact the case.